include/zephyr/drivers/rtc.h

```diff
@@ -527,6 +529,22 @@ static inline struct tm *rtc_time_to_tm(struct rtc_time *timeptr)
 	return (struct tm *)timeptr;
 }
 
+/**
+ * @brief Determine required calibration to 1 Hertz from frequency.
+ *
+ * @param frequency Frequency of the RTC in nano Hertz
+ *
+ * @return The required calibration in parts per billion
+ */
+static inline int32_t rtc_calibration_from_frequency(uint32_t frequency)
+{
+	if (frequency == 0) {
+		return INT32_MAX;
+	}
```

Comment:
This could like be changes to an ASSERT?

Reply:
Updated to use `__ASSERT_NO_MSG()`, additionally updated the header to include zephyr/kernel.h for the __ASSERT macros. (they are technically included through zephyr/device.h but relying on secondary includes seems improper)